src/legacy/core_plugins/data/public/search/aggs/agg_configs.ts

```diff
@@ -133,19 +145,18 @@ export class AggConfigs {
   }
 
   createAggConfig = <T extends AggConfig = AggConfig>(
-    params: AggConfig | AggConfigOptions,
+    params: Optional<AggConfigOptions, 'typesRegistry'>,
     { addToAggConfigs = true } = {}
   ) => {
-    let aggConfig;
-    if (params instanceof AggConfig) {
-      aggConfig = params;
-      params.parent = this;
-    } else {
-      aggConfig = new AggConfig(this, params);
-    }
+    const aggConfig = new AggConfig(this, {
+      typesRegistry: this.__typesRegistry,
```

Comment:
instead of passing in the registry lets find the correct type here and pass it into AggConfig